classes/search/ArticleSearchDAO.php

```diff
@@ -78,10 +78,10 @@ public function getPhraseResults($journal, $phrase, $publishedFrom = null, $publ
         }
 
         if (!empty($journal)) {
-            $sqlWhere .= ' AND i.journal_id = ?';
+            $sqlWhere .= ' AND (i.journal_id = ? OR i.journal_id is NULL)';
```

Comment:
I think this might leak publications between journals, because we're no longer checking the connection between publication and journal. You might want to verify the right journal by using `publications.submission_id` > `submissions.context_id` > `journals.journal_id` instead.

Reply:
that correct and this is a silly mistake as I was experimenting with different query . we can move this issue relation to journal as 
```sql
( (i.published = 1 AND i.journal_id = s.context_id)
                    OR ps.publication_id IS NOT NULL
                    OR p.issue_id IS NULL )
```
and make it based on submission's context_id like `s.context_id = ?` .